haystack_experimental/components/embedders/image/sentence_transformers_doc_image_embedder.py

```diff
@@ -335,11 +224,14 @@ def run(self, documents: List[Document]) -> Dict[str, List[Document]]:
                     "path": image_source_info["path"],
                     "page_number": page_number,
                 }
-                pdf_pages_info.append(pdf_page_info)
+                pdf_page_infos.append(pdf_page_info)
+            else:
+                # Process images directly
+                image: Union["Image", "ImageFile"] = PILImage.open(image_source_info["path"])
+                images_to_embed[doc_idx] = image
 
-        # Process PDF files and update images_to_embed
-        pdf_images = self._process_pdf_files(pdf_pages_info=pdf_pages_info)
-        for doc_idx, pil_image in pdf_images.items():
+        pdf_images_by_doc_idx = _batch_convert_pdf_pages_to_images(pdf_page_infos=pdf_page_infos, return_base64=False)
+        for doc_idx, pil_image in pdf_images_by_doc_idx.items():
             images_to_embed[doc_idx] = pil_image
 
         embeddings = self._embedding_backend.embed(
```

Comment:
Probably should have brought this up in a previous review, but at this point `images_to_embed` should no longer have any None's left in it right? Out of curiosity if there are any None's left would it cause the `embed` call to fail?